lib/pages/home_page.dart

```diff
@@ -610,10 +578,83 @@ class _HomePageState extends State<HomePage> with TickerProviderStateMixin {
     }
   }
 
+  void _onProjectSettingsChanged() {
+    _loadSettingsFromPrefs();
+    _saveProjectSettingsToFile(_projectDir!);
+  }
+
+  void _loadSettingsFromPrefs() {
+    setState(() {
+      _robotSize = Size(
+        widget.prefs.getDouble('robotWidth') ?? 0.75,
+        widget.prefs.getDouble('robotLength') ?? 1.0,
+      );
+      _holonomicMode = widget.prefs.getBool('holonomicMode') ?? false;
+      _generateJSON = widget.prefs.getBool('generateJSON') ?? false;
+      _generateCSV = widget.prefs.getBool('generateCSV') ?? false;
+      _pplibClient = widget.prefs.getBool('pplibClient') ?? false;
+
+      if (_pplibClient) {
+        PPLibClient.initialize(widget.prefs);
+      } else {
+        PPLibClient.stopServer();
+      }
+    });
+  }
+
+  void _loadProjectSettingsFromFile(Directory projectDir) {
+    File settingsFile = File(join(projectDir.path, _settingsDir));
+
+    if (settingsFile.existsSync()) {
+      try {
+        final fileContents = settingsFile.readAsStringSync();
+        final json = jsonDecode(fileContents);
+
+        widget.prefs
+            .setDouble('robotWidth', json['robotSize']?['width'] ?? 0.75);
+        widget.prefs
+            .setDouble('robotLength', json['robotSize']?['length'] ?? 1.0);
+        widget.prefs.setBool('holonomicMode', json['holonomicMode'] ?? false);
+        widget.prefs.setBool('generateJSON', json['generateJSON'] ?? false);
+        widget.prefs.setBool('generateCSV', json['generateCSV'] ?? false);
+      } catch (error) {
+        Log.error('An error occurred while loading settings');
+      }
+    }
+
+    _loadSettingsFromPrefs();
+  }
+
+  void _saveProjectSettingsToFile(Directory projectDir) {
+    File settingsFile = File(join(projectDir.path, _settingsDir));
+
+    if (!settingsFile.existsSync()) {
+      settingsFile.createSync(recursive: true);
+    }
+
+    const JsonEncoder encoder = JsonEncoder.withIndent('  ');
+
+    Map<String, dynamic> settings = {
+      'robotSize': {
+        'width': _robotSize.width,
+        'length': _robotSize.height,
+      },
+      'holonomicMode': _holonomicMode,
+      'generateJSON': _generateJSON,
+      'generateCSV': _generateCSV,
+    };
+
+    settingsFile.writeAsString(encoder.convert(settings)).then((_) {
+      Log.debug('Wrote json to file');
+    }).catchError((err) {
+      Log.error(err);
```

Comment:
For the debug log, could you make the message a bit more descriptive so its easy to know where it came from when seeing it in the console. Also, you should add a message to the error log so that the error object is the second argument and gets formatted correctly.

```suggestion
      Log.debug('Wrote project settings to file');
    }).catchError((err) {
      Log.error('Error writing project settings', err);
```